lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -121,39 +122,77 @@ public function convertException($message, DriverException $exception)
         return new Exception\DriverException($message, $exception);
     }
 
+
     /**
      * {@inheritdoc}
      */
     public function createDatabasePlatformForVersion($version)
     {
-        if ( ! preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $version, $versionParts)) {
-            throw DBALException::invalidPlatformVersionSpecified(
-                $version,
-                '<major_version>.<minor_version>.<patch_version>'
-            );
-        }
+
+        // Detect vendor and version
 
         if (false !== stripos($version, 'mariadb')) {
-            return $this->getDatabasePlatform();
-        }
+            // MariaDB does not follow one rule for versions ;)
 
-        $majorVersion = $versionParts['major'];
-        $minorVersion = isset($versionParts['minor']) ? $versionParts['minor'] : 0;
-        $patchVersion = isset($versionParts['patch']) ? $versionParts['patch'] : null;
 
-        if ('5' === $majorVersion && '7' === $minorVersion && null === $patchVersion) {
-            $patchVersion = '9';
-        }
 
-        $version = $majorVersion . '.' . $minorVersion . '.' . $patchVersion;
+            // Will catch things like '10.2.8-MariaDB-1~lenny-log'
+            preg_match('/(?P<major>\d{1,2})\.(?P<minor>\d{1,2})\.(?P<patch>\d+)/', $version, $firstMatch);
```

Comment:
Why restrict the number of digits to 2 in `major` and `minor` but not in the `patch`? `\d` looks reasonable to me in all cases.